### ISSUE TITLE
Training with weight bitwidth setting. 

### ISSUE THREAD

**Issue by donghn** [zzzxxxttt/pytorch_DoReFaNet#9]
Hi.
I trained the Alexnet with 1bit weight it is fine. But when I train with 2 or 4 bit weight, the accuracy is 10% and does not change during training and loss is around 7.01. Maybe, There is a problem with nn.round function. 
Do anyone know how to solve it?
Thank you very much.

**Comment by zzzxxxttt** [comment]
Did you train from scratch or finetune from a pretrained model?

**Comment by donghn** [comment]
I trained the Alexnet model from scratch. I changed the learning_rate = 0.0001 or 0.001 or 0.01. But it seems nothing happen during training.

**Comment by zzzxxxttt** [comment]
I recommend finetune a pretrained model, and I'll do some experiments to check the code.

**Comment by donghn** [comment]
thank you @zzzxxxttt. Please share your experiments when you finish.

**Comment by zzzxxxttt** [comment]
Hi @donghn, after a close look at the quantization code, I found there is a bug when bitwidth>1, the weight tensor is not scaled properly. It is now fixed in my new commit, I trained for 5 epochs from scratch and it converges now.
